### PR TITLE
Implement RFID free login

### PR DESCRIPTION
**Pull request by emanuelen5**
# TODO
- [x] PIN login
- [x] Remove tag reader class
- [x] Remove all references of GUI states for tag reader
- [x] Remove events that relates to tag reader
- [x] ...

### CODE REVIEW

**Review by InMyOrbit — COMMENTED** [review]
Changes already made in new commit.

File: src/backend/member.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 logger = getLogger("memberbooth")
 
 
+# TODO I guess we can remove this.
```
Comment (emanuelen5) [inline src/backend/member.py:10]: Yes, remove this and all places it is used

File: src/gui/design.py [hunk 2]
```diff
@@ -13,6 +11,9 @@
 MAX_DESCRIPTION_LENGTH = 256
 TIMEOUT_TIMER_PERIOD_MS = 60 * 1000
 TEMPORARY_STORAGE_LABEL_DEFAULT_TEXT = 'Describe what you want to store here...'
+MEMBER_NUMBER_LENGTH = 4
+#PIN_CODE_LENGTH = 6
```
Comment (emanuelen5) [inline src/gui/design.py:15]: It should be `MEMBER_NUMBER_MINIMUM_LENGTH`. In about 10 years time we will have more than 10000 unique members, and thus more than 4 numbers for the member number :)

File: src/gui/states.py [hunk 3]
```diff
@@ -169,9 +135,17 @@ def on_event(self, event):
...
+            try:
+                member_number = event.data
+                logger.debug(f"Pin code requested for member number = {member_number} - implement here")
+            except NoMatchingMemberNumber:
+                self.gui.reset_gui()
+                self.gui.show_error_message("There is no member with that member number.")
+            except Exception as e:
+                logger.exception("Unexpected exception")
+                self.gui.show_error_message(f"Error... \n{e}")
+                self.gui.reset_gui()
```
Comment (emanuelen5) [inline src/gui/states.py:148]: `PIN_CODE_REQUESTED` används inte?

File: src/gui/states.py [hunk 4]
```diff
@@ -34,8 +27,9 @@ def change_state(self):
 
     def gui_callback(self, gui_event):
 
+        # TODO What does this do?
```
Comment (emanuelen5) [inline src/gui/states.py:30]: Doesn't the comment below explain it? :)

File: src/gui/states.py [hunk 5]
```diff
@@ -96,70 +90,42 @@ class WaitingState(State):
...
-                return MemberIdentified(self.application, self.master, self.member)
-            except NoMatchingTagId:
+                login = event.data
+                member = Member.from_member_number(self.application.makeradmin_client, login.member_number)
+                logger.debug(f"Login requested with member_numer = {login.member_number}, pin_code = {login.pin_code} - implement here")
+                return MemberIdentified(self.application, self.master, member)
+            except NoMatchingMemberNumber:
                 self.gui.reset_gui()
-                self.gui.show_error_message("Could not find a member that matches the specific tag")
+                self.gui.show_error_message("Login incorrect.")
```
Comment (emanuelen5) [inline src/gui/states.py:125]: I don't think this exception can be thrown here. Either the login works or we get and `IncorrectPinCode` exception. The backend never tells the memberbooth why it did not work.

**Comment by emanuelen5** [comment]
I merged it with master so there are no conflicts anymore

**Pull request merged**